bam_fastq.c

```diff
@@ -0,0 +1,1019 @@
+    state->flag_on = opts->flag_on;
+    state->flag_off = opts->flag_off;
+    state->flag_alloff = opts->flag_alloff;
+    state->has12 = opts->has12;
+    state->use_oq = opts->use_oq;
+    state->illumina_tag = opts->illumina_tag;
+    state->copy_tags = opts->copy_tags;
+    state->filetype = opts->filetype;
+    state->def_qual = opts->def_qual;
+    state->index_sequence = NULL;
+    state->hstdout = NULL;
+    state->compression_level = opts->compression_level;
+
+    state->taglist = kl_init(ktaglist);
+    if (opts->extra_tags) {
+        char *save_p;
+        char *s = strtok_r(opts->extra_tags, ",", &save_p);
+        while (s) {
+            if (strlen(s) != 2) {
+                fprintf(stderr, "Parsing extra tags - '%s' is not two characters\n", s);
+                free(state);
+                return false;
+            }
+            char **et = kl_pushp(ktaglist, state->taglist);
+            *et = s;
+            s = strtok_r(NULL, ",", &save_p);
+        }
+    }
+
+    state->fp = sam_open(opts->fn_input, "r");
+    if (state->fp == NULL) {
+        print_error_errno("bam2fq","Cannot read file \"%s\"", opts->fn_input);
+        free(state);
+        return false;
+    }
+
+    state->p.pool = NULL;
+    if (opts->ga.nthreads > 0) {
+        if (!(state->p.pool = hts_tpool_init(opts->ga.nthreads))) {
+            fprintf(stderr, "Failed to create thread pool\n");
+            free(state);
+            return false;
+        }
+        state->p.qsize = opts->ga.nthreads*2;
+        hts_set_thread_pool(state->fp, &state->p);
+    }
+
+    uint32_t rf = SAM_QNAME | SAM_FLAG | SAM_SEQ | SAM_QUAL;
+    if (opts->use_oq || opts->extra_tags || opts->index_file[0]) rf |= SAM_AUX;
+    if (hts_set_opt(state->fp, CRAM_OPT_REQUIRED_FIELDS, rf)) {
+        fprintf(stderr, "Failed to set CRAM_OPT_REQUIRED_FIELDS value\n");
+        free(state);
+        return false;
+    }
+    if (hts_set_opt(state->fp, CRAM_OPT_DECODE_MD, 0)) {
+        fprintf(stderr, "Failed to set CRAM_OPT_DECODE_MD value\n");
+        free(state);
+        return false;
+    }
+    if (opts->fnse) {
+        state->fpse = open_fqfile(opts->fnse, state->compression_level, &state->p);
+        if (state->fpse == NULL) {
+            print_error_errno("bam2fq", "Cannot write to singleton file \"%s\"", opts->fnse);
+            free(state);
+            return false;
+        }
+    }
+
+    if (opts->ga.reference) {
+        if (hts_set_fai_filename(state->fp, opts->ga.reference) != 0) {
+            print_error_errno("bam2fq", "cannot load reference \"%s\"", opts->ga.reference);
+            free(state);
+            return false;
+        }
+    }
+
+    int i, j;
+    for (i = 0; i < 3; ++i) {
+        if (opts->fnr[i]) {
+	    for (j = 0; j < i; j++)
+		if (opts->fnr[j] && strcmp(opts->fnr[j], opts->fnr[i]) == 0)
+		    break;
+	    if (j == i) {
+		state->fpr[i] = open_fqfile(opts->fnr[i], state->compression_level, &state->p);
```

Comment:
Please replace the tabs with spaces.

Reply:
Good spot.  Fixed.

It seems this is a result of upgrading my OS causing emacs on the old servers to no longer work properly (it dislikes something in my upgraded config apparently).  Grr.